### PR TITLE
LsThing advanced search return flat structure

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
New "flat" output format for LsThing advanced search.  Allows you to specify return values, labels and thing primary attributes (codeName, id, recordedBy, lsType...etc.) to return.

## Related Issue
ACAS-460

## How Has This Been Tested?
ACAS client tests and manual testing of Example thing browser with 10,000 things. 10,000 search/render in about 10-15 seconds on my local machine.

### CODE REVIEW

File: src/main/java/com/labsynch/labseer/dto/LsThingQueryResultDTO.java [hunk 1]
```diff
@@ -89,6 +98,138 @@ public static Collection<LsThingQueryResultDTO> fromJsonArrayToLsThingQueryResul
...
+                    // Make sure this isn't a duplicate key
+                    if(resultObject.has(key)) {
+                        throw new RuntimeException("Key '" + key + "' is a duplicate. All keys must be unique.");
+                    }
+                }
+
+                // If the state type/kind and value type/kind are not null then it's a value return listing
+                if(returnQuery.getStateType() != null && returnQuery.getStateKind() != null && returnQuery.getValueType() != null && returnQuery.getValueKind() != null) {
+                    // Get the first matching value from the thing that is not ignored
+                    LsThingValue v = lsThing.getLsStates().stream()
```
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/dto/LsThingQueryResultDTO.java:177]: Very cool!

**Review by bffrost — CHANGES_REQUESTED** [review]
Main code LGTM. Noted a few peripheral things that potentially looked like accidental checkins. Please address, then should be good to go.

File: src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -18,6 +18,11 @@
     <!-- <Logger name="org.hibernate.engine.transaction.internal.TransactionImpl" level="debug"/>
     <Logger name="org.springframework.transaction" level="debug"/>
     <Logger name="org.springframework.orm.jpa" level="debug"/> -->
+
+    <!-- Log all SQL statements -->
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply src/main/resources/log4j2.xml]: No, thanks, just removed.

File: pom.xml [hunk 3]
```diff
@@ -87,7 +87,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.]]>
 		<profile>
 			<id>bbchem</id>
 			<activation>
-				<activeByDefault>false</activeByDefault>
+				<activeByDefault>true</activeByDefault>
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply pom.xml]: Unintended, just reverted.

**Review by bffrost — APPROVED** [review]
LGTM

**Pull request merged**